XiEditor/AppDelegate.swift

```diff
@@ -286,6 +288,14 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
             }
         }
     }
+    
+    func languageChanged(viewIdentifier: String, languageId: String) {
+        DispatchQueue.main.async { [weak self] in
+            self?.languageId = languageId
+            let document = self?.documentForViewIdentifier(viewIdentifier: viewIdentifier)
```

Comment:
actually, what's going on here? Is there a reason we need to store the `languageId` at all outside of the particular view that's effected?

Reply:
You're right, that's weird. This shouldn't be in the code.